### ISSUE TITLE
Test failures, cannot find module artifact

### ISSUE THREAD

**Issue by lucaswerkmeister** [ceylon/ceylon-compiler#2224]
When running the compiler tests (`ant test`), I get the following error 12 times:

> Compilation failed ERROR: frontend: 25: cannot find module artifact org.hibernate.javax.persistence:hibernate-jpa-2.1-api-1.0.0.Final(.car|.jar) - dependency tree: com.redhat.ceylon.compiler.java.test.metamodel/123 -> org.hibernate.javax.persistence:hibernate-jpa-2.1-api/1.0.0.Final

and also this one (once, `testMdlCeylonAetherDuplicateImports`):

> Unexpected error: ERROR: frontend: 24: cannot find module artifact org.apache.httpcomponents:httpclient-4.3.2(.car|.jar) - dependency tree: com.redhat.ceylon.compiler.java.test.cmr.modules.ceylonAetherDuplicateImports/1 -> org.apache.httpcomponents:httpclient/4.3.2

**Comment by FroMage** [comment]
Damn there are a lot of tests not passing, not just this one.

**Comment by lucaswerkmeister** [comment]
It seems there are a lot more tests failing on the CLI than in the IDE.

**Comment by FroMage** [comment]
Yes, but I'm fixing that too. They should be the same.

**Comment by FroMage** [comment]
In theory I've fixed it, can you confirm?

**Comment by lucaswerkmeister** [comment]
A few failures seem to be gone (50 to 44), but I’m still getting the “cannot find module artifact” ones.

**Comment by FroMage** [comment]
This is not normal. I added the maven-support jar to the classpath so it should "just work". Can you tweak the `test.all` target in `ceylon-compiler/build.xml` to run only a single of such failing test and give me the stderr/stdout output? There's a `test` task commented out to help you run a single test, just uncomment it, and comment `batchtest`.

**Comment by lucaswerkmeister** [comment]
I ran `MetamodelTests.testBug1205`, and it failed without printing anything (unless Ant hid it). Is there some setting to increase verbosity?

**Comment by FroMage** [comment]
Ant hides it, look into the reports and find the `stderr/stdout` link on the bottom right.

**Comment by lucaswerkmeister** [comment]
OIC, thanks.
Stderr:
> WARNING:It looks like you are using class files from a Java newer than 1.7.
  Everything should work well, but if not, let us know at https://github.com/ceylon/ceylon-compiler/issues.
  In the near future, the Ceylon compiler will be upgraded to handle Java 1.8.
test/src/com/redhat/ceylon/compiler/java/test/metamodel/module.ceylon:25:ERROR:cannot find module artifact org.hibernate.javax.persistence:hibernate-jpa-2.1-api-1.0.0.Final(.car|.jar)
  	- dependency tree: com.redhat.ceylon.compiler.java.test.metamodel/123 -> org.hibernate.javax.persistence:hibernate-jpa-2.1-api/1.0.0.Final

No out listed, presumably empty

Full error:
> Compilation failed ERROR: frontend: 25: cannot find module artifact org.hibernate.javax.persistence:hibernate-jpa-2.1-api-1.0.0.Final(.car|.jar) - dependency tree: com.redhat.ceylon.compiler.java.test.metamodel/123 -> org.hibernate.javax.persistence:hibernate-jpa-2.1-api/1.0.0.Final

junit.framework.AssertionFailedError: Compilation failed
ERROR: frontend: 25: cannot find module artifact org.hibernate.javax.persistence:hibernate-jpa-2.1-api-1..0.0.Final(.car|.jar)
- dependency tree: com.redhat.ceylon.compiler.java.test.metamodel/123 -> org.hibernate.javax.persistence:hibernate-jpa-2.1-api/1.0.0.Final
at com.redhat.ceylon.compiler.java.test.CompilerTests.assertCompilesOk(CompilerTests.jaava:512)
at com.redhat.ceylon.compiler.java.test.CompilerTests.compareWithJavaSource(CompilerTests.java:477)

at com.redhat.ceylon.compiler.java.test.CompilerTTests.compareWithJavaSource(CompilerTests.java:444)
at com.redhat.ceylon.compiler.java.test.CompilerTests.compareWithJavaSource(CompilerTests.java:199)
at com.redhat.ceylon.compiler.java.test.metamodel..MetamodelTests.testBug1205(MetamodelTests.java:67)

TLDR: Nothing new :/

**Comment by FroMage** [comment]
And if you run it from Eclipse, does it work?

**Comment by FroMage** [comment]
It could be a `settings.xml` file in your Maven repo that's breaking stuff.

**Comment by lucaswerkmeister** [comment]
I’ll check, gotta update first though.

**Comment by lucaswerkmeister** [comment]
Weird, now I’m getting an error “cannot find javac resource bundle for locale en_US”. It worked a few days ago though.

**Comment by lucaswerkmeister** [comment]
> It could be a settings.xml file in your Maven repo that's breaking stuff.

No file by that name in my `~/.m2` directory. Could I just completely delete it, or would that break stuff? (AFAIK I don’t use anything else with Maven.)

**Comment by FroMage** [comment]
> Weird, now I’m getting an error “cannot find javac resource bundle for locale en_US”. It worked a few days ago though.

Running it in Eclipse?

**Comment by lucaswerkmeister** [comment]
Yes.

**Comment by lucaswerkmeister** [comment]
On all tests btw.

**Comment by lucaswerkmeister** [comment]
Looks like a clean rebuild in the IDE fixed that…

**Comment by lucaswerkmeister** [comment]
…and after JUnit is done running 2250/2243 tests (sure), I can report that indeed I don’t these errors from Eclipse.

**Comment by FroMage** [comment]
OK, so I'll need you to run ant verbosely then for that single test.

**Comment by lucaswerkmeister** [comment]
https://gist.github.com/lucaswerkmeister/f4fbf1effb02ca083ea0

**Comment by lucaswerkmeister** [comment]
Oh, and just in case it matters, there’s no network activity while the test runs.

**Comment by FroMage** [comment]
> dropping /home/lucas/.ceylon/repo/com/redhat/ceylon/maven-support/${module.com.redhat.ceylon.maven-support.version}/com.redhat.ceylon.maven-support-${module.com.redhat.ceylon.maven-support.version}.jar from path as it doesn't exist

That's the winner.

**Comment by lucaswerkmeister** [comment]
So it is a problem with my local install? A missing jar?

**Comment by lucaswerkmeister** [comment]
File `/home/lucas/.ceylon/repo/com/redhat/ceylon/maven-support/2.0/com.redhat.ceylon.maven-support-2.0.jar` exists though.

**Comment by lucaswerkmeister** [comment]
um… shouldn’t we see some actual version number instead of `${module.com.redhat.ceylon.maven-support.version}` in that message? :D

**Comment by FroMage** [comment]
Lemme finish merging your PR and I'll look at this.

**Comment by lucaswerkmeister** [comment]
Ok, sure.

**Comment by FroMage** [comment]
so `module.com.redhat.ceylon.maven-support.version` comes from `ceylon-common/common.properties`. Do you see it there?

**Comment by lucaswerkmeister** [comment]
I only have a ceylon-common/common**-build**.properties, and it doesn’t include that version. It is however present in `ceylon-runtime`’s `build.properties`, and used in `build.xml`.

**Comment by FroMage** [comment]
Because I'm an idiot, sorry. It's pushed now.

**Comment by lucaswerkmeister** [comment]
Yay, it works! Only 13 fails and 19 errors now! Thanks! :)

**Comment by FroMage** [comment]
I get:

    [junit] Tests run: 2672, Failures: 5, Errors: 13, Skipped: 56, Time elapsed: 178.279 sec

So can you give me your test report?

**Comment by lucaswerkmeister** [comment]
Sure! [fails](https://lucaswerkmeister.de/ceylon-compiler-test-reports/alltests-fails.html)<sup>[archived](https://web.archive.org/web/20150707184935/http://lucaswerkmeister.de/ceylon-compiler-test-reports/alltests-fails.html)</sup>, [errors](https://lucaswerkmeister.de/ceylon-compiler-test-reports/alltests-errors.html)<sup>[archived](https://web.archive.org/web/20150707185012/http://lucaswerkmeister.de/ceylon-compiler-test-reports/alltests-errors.html)</sup>.

And both the completion tests and ceylondoc tests have been failing for me since the beginning of time, so this isn’t new breakage that you’d need to worry about.

**Comment by FroMage** [comment]
So I fixed a few more tests. The only one remaining which is not supposed to fail, but fails if you don't have a local Maven repo is the aether completion/search test because Maven central is broken again and gives us metadata with bogus checksums. Works on my machine because I had a metadata file located and for some reason it did not download a new one. I'll just find another module to test on, with (currently) non-broken metadata :(

**Comment by FroMage** [comment]
OK, if you try now you should have exactly the same results on the CLI and in Eclipse. Reopen if not ;)

**Comment by lucaswerkmeister** [comment]
Yup, 6 fails 7 errors in both :) thanks!
